tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,53 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// The default maximum number of buffered log lines.
+///
+/// If [`NonBlocking`][non-blocking] is lossy, it will drop spans/events at capacity.
+/// capacity. If [`NonBlocking`][non-blocking] is _not_ lossy,
+/// backpressure will be exerted on senders, causing them to block their
+/// respective threads until there is available capacity.
+///
+/// [non-blocking]: ./struct.NonBlocking.html
+/// Recommended to be a power of 2.
 pub const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
 
+/// A guard that flushes an associated [`NonBlocking`] of spans/events on drop.
+///
+/// Writing to a [`NonBlocking`] writer will **not** immediately write a span or event to the underlying
+/// output. Instead, the span or event will be written by a dedicated logging thread at some later point.
+/// To increase throughput, the non-blocking writer will flush to the underlying output on
+/// a periodic basis rather than every time a span or event is written. This means that if the program
+/// terminates abruptly (such as through an uncaught `panic` or a `std::process::exit`), some spans
+/// or events may not be written.
+///
+/// [`NonBlocking`]: ./struct.NonBlocking.html
+/// Since spans/events and events recorded near a crash are often necessary for diagnosing the failure,
+/// `WorkerGuard` provides a mechanism to ensure that _all_ buffered logs are flushed to their output.
+/// `WorkerGuard` should be assigned in the `main` function or whatever the entrypoint of the program is.
+/// This will ensure that the guard will be dropped during an unwinding or when `main` exits
+/// successfully.
```

Comment:
Feels like it would be helpful to have an example of this.

Reply:
added